lib/ts/recipe/thirdparty/components/library/providerButton.tsx

```diff
@@ -21,9 +23,69 @@ export type ProviderButtonProps = {
     logo?: JSX.Element;
 };
 
-export default function ProviderButton({ logo, providerName, displayName }: ProviderButtonProps) {
+export default function ProviderButton({ logo, providerName, displayName }: ProviderButtonProps): React.ReactElement {
     const t = useTranslation();
     const providerStyleName = `provider${providerName}`;
+    const buttonTextContainerRef = useRef<HTMLDivElement>(null);
+    const buttonTextRef = useRef<HTMLSpanElement>(null);
+    const SCROLL_ANIMATION_CLASS = "scroll-text-animation";
+
+    const isTextOverflowing = (element: Element) => {
+        return element.scrollWidth > element.clientWidth;
+    };
+
+    function startAnimation() {
+        const buttonTextContainer = buttonTextContainerRef.current;
+        const buttonText = buttonTextRef.current;
+        const BUFFER = 10;
+
+        if (!buttonTextContainer || !buttonText || !buttonTextContainer.classList.contains(SCROLL_ANIMATION_CLASS)) {
+            return;
+        }
+
+        const keyframes = [
+            { transform: "translateX(0px)" },
+            {
+                transform: `translateX(-${
+                    buttonTextContainer.scrollWidth - buttonTextContainer.clientWidth + BUFFER
+                }px)`,
+            },
+        ];
+        const options: KeyframeAnimationOptions = {
+            duration: 3000,
+            iterations: 1,
+        };
+        buttonText.style.display = "inline-block";
+        buttonText.animate(keyframes, options);
+    }
+
+    const stopAnimation = () => {
+        const buttonText = buttonTextRef.current;
+        if (!buttonText) {
+            return;
+        }
+        buttonText.getAnimations().forEach((animation) => animation.cancel());
+    };
```

Comment:
why do we need these instead of applying the animation on hover using css?

Reply:
Based on this comment:

`Is there a way to make the animation stop (or reverse) when the text reaches the end?
On my machine it'll scroll the provider name off the screen then go back to the start. Instead I'd prefer if it stopped once the right edge of the text is on the screen (plus some margin maybe)`.

The animation scroll distance is dynamic (depends on the length of the overflowing text). Since this is calculated using javascript, I decided to apply the animation using the web animation API.

Reply:
I much prefer a pure css solution. Try this approach: https://codepen.io/porcellus/pen/wvLxzmN

Reply:
Got it. that works too. I have updated the code